src/android/com/iotum/CordovaAndroidScreenshare.java

```diff
@@ -300,15 +308,26 @@ public void onDestroy() {
         mTimer = null;
       }
       sMediaProjection.stop();
+      // Stop the foreground service
+      Activity activity = cordova.getActivity();
+      Intent intent = new Intent(activity, MediaProjectionService.class);
+      intent.setAction("stop");
+      activity.getApplicationContext().startService(intent);
     }
   }
 
   /******************************************
    * UI Widget Callbacks
    *******************************/
   private void startProjection() {
+    // Request permission to capture screen
     cordova.setActivityResultCallback(this);
     cordova.startActivityForResult(this, mProjectionManager.createScreenCaptureIntent(), REQUEST_CODE);
+    // Start a mandatory MediaProjection foreground service with a notification for Android SDK 29
+    Activity activity = cordova.getActivity();
+    Intent intent = new Intent(activity, MediaProjectionService.class);
+    intent.setAction("start");
+    activity.getApplicationContext().startService(intent);
```

Comment:
Unfortunately I couldn't start the foreground service after getting the screenshare permissions, so it must be started beforehand.

This means I needed to stop the service if the permissions were rejected.